src/alfasim_sdk/result_reader/reader.py

```diff
@@ -295,3 +330,154 @@ def get_profile_metadata(meta: Dict) -> ProfileMetadata:
             get_profile_metadata(profile_metadata)
             for profile_metadata in metadata.profiles.values()
         ]
+
+
+def _non_empty_dict_validator(values_type: type) -> Callable:
+    def validator(inst: Any, attribute: attr.Attribute, value: Any) -> None:
+        attr.validators.min_len(1)(inst, attribute, value)
+        attr.validators.instance_of(dict)(inst, attribute, value)
+        assert isinstance(value, dict)
+
+        some_dict_value = list(value.values())[0]
+        if not isinstance(some_dict_value, values_type):
+            raise ValueError(
+                f"values of dict attribute `{attribute.name}` should be of type {values_type},"
+                f" received: {type(some_dict_value)}"
+            )
+
+    return validator
+
+
+def _non_empty_attr_validator(attr_name: str) -> Callable:
+    def validator(inst: Any, attribute: attr.Attribute, value: Any) -> None:
+        attr.validators.min_len(1)(inst, attribute, getattr(value, attr_name))
+
+    return validator
+
+
+@define(frozen=True)
+class GlobalSensitivityAnalysisResults:
+    timeset: np.ndarray = attr.field(validator=attr.validators.min_len(1))
+    coefficients: dict[GSAOutputKey, np.ndarray] = attr.field(
+        validator=attr.validators.min_len(1)
+    )
+    metadata: GlobalSensitivityAnalysisMetadata = attr.field(
+        validator=_non_empty_attr_validator("items")
+    )
+
+    @classmethod
+    def from_directory(cls, result_dir: Path) -> Self | None:
+        metadata = read_global_sensitivity_analysis_meta_data(result_dir)
+        if metadata is None:
+            return None
+
+        return cls(
+            timeset=read_uq_time_set(
+                result_dir, GLOBAL_SENSITIVITY_ANALYSIS_GROUP_NAME
+            ),
+            coefficients=read_global_sensitivity_coefficients(metadata),
+            metadata=metadata,
+        )
+
+    def get_sensibility_curve(
```

Comment:
All this feature is called ```Global Sensitivity Analysis```, i believe that the correct name for this functions is:
```suggestion
    def get_sensitivity_curve(
```

Reply:
Although "sensitivity" and "sensibility" convey the same meaning, I agree that it's better to be consistent. Will change that, good catch.